### ISSUE TITLE
Questions about GPU time-sharing on Kubernetes

### ISSUE THREAD

**Issue by jxl4650152** [NVIDIA/k8s-device-plugin#404]
### 1. Issue or feature description
Is it possible to enable time-sharing on select GPUs within a single node, rather than all of them?
If not all GPUs on a single node support time-slicing, what behavior can be expected from the plugin?

**Comment by elezar** [comment]
The device plugin does not currently have a mechanism to expose different devices as different resource types. It is also not possible to apply sharing settings on a per-GPU basis.

Note that the plugin will still allow a sharing setting to be applied to GPUs that may not support this feature and effectively reports the same device multiple times to the Kubelet. The behaviour of containers that are both started on a device where time-slicing is not supported will depend on the application and the device, but should mirror what happens when two applications which access the same device are started on the host.

**Comment by jxl4650152** [comment]
Thank you for your response. We now have a clear understanding of the limitations of this plugin. We are using this repo for managing GPUs on Kubernetes, and we have also noticed another [repo](https://gitlab.com/nvidia/cloud-native/k8s-dra-driver) related to DRA（Dynamic Resource Allocation）, which offers greater flexibility and richer functionality. Regarding this, we have another question: Will there be strong support for using GPUs on Kubernetes through the DRA approach in the future?

**Comment by klueska** [comment]
Yes DRA will be well supported. We see it as the future of GPU support in Kubernetes.

Please see:
https://m.youtube.com/watch?v=_fi9asserLE&ab_channel=CNCF%5BCloudNativeComputingFoundation%5D

**Comment by frittentheke** [comment]
Like @klueska said this is not (yet possible), see https://github.com/NVIDIA/k8s-device-plugin/blob/35ad18080eded1889dc1eaee1132debddfd6757c/api/config/v1/replicas.go#L61C5-L61C16

I myself would also like to only enable time-slicing on a subset of GPUs. Is there any chance we could make this issue here a feature request @elezar? Or is this capability never coming to the device plugin in favor of DRA?

**Comment by shivamerla** [comment]
As @klueska [commented](https://github.com/NVIDIA/k8s-device-plugin/issues/404#issuecomment-1574742288), this is supported with DRA (which is GA from k8s 1.34). Uses can share the same resource claim across multiple workload pods to achieve this.
